lib/jsonapi-serializers/serializer.rb

```diff
@@ -320,19 +320,20 @@ def self.serialize_primary(object, options = {})
       return if object.nil?
 
       serializer = serializer_class.new(object, options)
-      data = {
+
+      {
         'id' => serializer.id.to_s,
         'type' => serializer.type.to_s,
         'attributes' => serializer.attributes,
-      }
-
-      # Merge in optional top-level members if they are non-nil.
-      # http://jsonapi.org/format/#document-structure-resource-objects
-      data.merge!({'attributes' => serializer.attributes}) if !serializer.attributes.nil?
-      data.merge!({'links' => serializer.links}) if !serializer.links.nil?
-      data.merge!({'relationships' => serializer.relationships}) unless serializer.relationships.empty?
-      data.merge!({'meta' => serializer.meta}) if !serializer.meta.nil?
-      data
+      }.tap do |data|
```

Comment:
There's no need to `tap` here because it's not a method chain. For clarity and personal style reasons (I like very clear returns), please keep `data` assigned to the hash and then do your `each` block to modify it, then return it.